lib/src/kbasesearchengine/main/NarrativeInfoDecorator.java

```diff
@@ -96,51 +107,69 @@ public SearchObjectsOutput searchObjects(final SearchObjectsInput params, final
     public GetObjectsOutput getObjects(final GetObjectsInput params, final String user)
             throws Exception {
         GetObjectsOutput getObjsOutput = searchInterface.getObjects(params, user);
+        if (getObjsOutput.getAdditionalProperties().get(removedGuids) == null &&
+                "true".equals(System.getProperty(removedGuidsEnv))){
+            getObjsOutput.getAdditionalProperties().put(removedGuids, new ArrayList<>());
+        }
+        final List<String> inputArr = "true".equals(System.getProperty(removedGuidsEnv)) ?
+                (List<String>) getObjsOutput.getAdditionalProperties().get(removedGuids) : new ArrayList<>();
         if (params.getPostProcessing() != null) {
             if (params.getPostProcessing().getAddNarrativeInfo() != null &&
                     params.getPostProcessing().getAddNarrativeInfo() == 1) {
                 getObjsOutput = getObjsOutput.withAccessGroupNarrativeInfo(addNarrativeInfo(
                         getObjsOutput.getObjects(),
-                        getObjsOutput.getAccessGroupNarrativeInfo()));
+                        getObjsOutput.getAccessGroupNarrativeInfo(),
+                        inputArr));
             }
         }
         return getObjsOutput;
     }
 
+    /**
+     * Adds narrative information for non deleted and valid narrative workspaces. Removes results otherwise and
+     * add the removed result's guid to removedGuids list, if list is not null.
```

Comment:
I don't think the list can be null can it?

Reply:
ops didn't update this comment from when we changed to not do nulls.